Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -467,7 +467,6 @@ void Connection::invalidate()
     m_isValid = false;
     if (!m_client)
         return;
-    assertIsCurrent(dispatcher());
```

Comment:
In general this cannot be relaxed without further code ensuring the correctness.
E.g. in practice it must be guaranteed that m_client is set and removed from the same thread, so there's no cross-thread access to this unlocked variable. So open and invalidate must be done from the same thread.
Which is what the linked pr tries to do ..